Sources/SwiftProtobuf/Google_Protobuf_Timestamp_Extensions.swift

```diff
@@ -270,49 +266,53 @@ public extension Google_Protobuf_Timestamp {
     public init(decoder: inout JSONDecoder) throws {
         self.init()
         let s = try decoder.scanner.nextQuotedString()
-        let timestamp = try parseTimestamp(s: s)
-        seconds = timestamp.0
-        nanos = timestamp.1
+        (seconds, nanos) = try parseTimestamp(s: s)
     }
 
     public func serializeJSON() throws -> String {
-        let s = seconds
-        let n = nanos
-        if let formatted = formatTimestamp(seconds: s, nanos: n) {
+        if let formatted = formatTimestamp(seconds: seconds, nanos: nanos) {
             return "\"\(formatted)\""
         } else {
             throw EncodingError.timestampJSONRange
         }
     }
 
-    func serializeAnyJSON() throws -> String {
+    public func serializeAnyJSON() throws -> String {
         let value = try serializeJSON()
         return "{\"@type\":\"\(anyTypeURL)\",\"value\":\(value)}"
     }
 }
 
-private func normalizedTimestamp(seconds: Int64, nanos: Int32) -> Google_Protobuf_Timestamp {
-    var s = seconds
-    var n = nanos
-    if n >= 1000000000 || n <= -1000000000 {
-        s += Int64(n) / 1000000000
-        n = n % 1000000000
+private func normalizeForTimestamp(seconds: Int64, nanos: Int32) -> (seconds: Int64, nanos: Int32) {
+    // The Timestamp spec says that nanos must be in the range [0, 999999999), as in actual
+    // modular arithmetic. Integer % and / do not reflect this, so mod() and div() are defined
+    // for this purpose. Factored out, they make normalization very straightforward.
+    func mod(_ a: Int32, _ b: Int32) -> Int32 {
+        assert(b > 0)
+        return a % b >= 0 ? a % b : a % b + b
     }
-    if s > 0 && n < 0 {
-        n += 1000000000
-        s -= 1
-    } else if s < 0 && n > 0 {
-        n -= 1000000000
-        s += 1
+    func div(_ a: Int32, _ b: Int32) -> Int32 {
```

Comment:
Yeah it does, but then I need to put the cast into the return expression for div, which is ... not right.

I'm still getting used to Swift's refusal to allow for implicit conversions for integer types when they ought to be safe. Or, in this case, I would like to see an operator + with signature (Int64, Int32) -> Int64. It makes me wonder if I'm thinking about something wrong.